File: src/main/java/io/xol/chunkstories/api/input/KeyboardKeyInput.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.input;

/**
 * Describes a key assignated to some action
 */
public interface KeyboardKeyInput extends Input {
	/**
	 * Returns the name of the bind
	 */
	@Override
	public String getName();

	/**
	 * Returns true if the key is pressed
	 */
	@Override
	public boolean isPressed();
}

File: src/main/java/xyz/chunkstories/api/input/KeyboardKeyInput.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.input

/** Describes a key assignated to some action  */
interface KeyboardKeyInput : Input {
    /** Returns the name of the bind  */
    override val name: String

    /** Returns true if the key is pressed  */
    override val isPressed: Boolean
}
